hydromt_sfincs/sfincs.py

```diff
@@ -1316,6 +1317,49 @@ def setup_observation_points(
         self.set_geoms(gdf_obs, name)
         self.set_config(f"{name}file", f"sfincs.{name}")
 
+    def setup_observation_lines(
+        self,
+        locations: Union[str, Path, gpd.GeoDataFrame],
+        merge: bool = True,
+        **kwargs,
+    ):
+        """Setup model observation lines (cross-sections) to monitor discharges.
+
+        Adds model layers:
+
+        * **crs** geom: observation lines (cross-sections)
+
+        Parameters
+        ---------
+        locations: str, Path, gpd.GeoDataFrame, optional
+            Path, data source name, or geopandas object for observation lines (cross-sections).
+        merge: bool, optional
+            If True, merge the new observation lines with the existing ones. By default True.
+        """
+        name = self._GEOMS["observation_lines"]
+
+        # FIXME ensure the catalog is loaded before adding any new entries
+        self.data_catalog.sources
+
+        # FIXME assert_gtype="LineString" does not work for MultiLineString and default seems to be Point (??)
+        gdf_obs = self.data_catalog.get_geodataframe(
+            locations, geom=self.region, assert_gtype=None, **kwargs
+        ).to_crs(self.crs)
+
+        # make sure MultiLineString are converted to LineString
+        gdf_obs = gdf_obs.explode().reset_index(drop=True)
```

Comment:
The original data indices are ignored here. Is it important that the indices of the original data are kept? I would think that it might be useful to match indices in the results to your input data. I noticed that also in obs the indices get 'lost' when reading or merging. 

Not necessarily something we need to solve here and now, but perhaps good to discuss in conjunction with #115

Reply:
Fair point, since we give the observation points and the observation_lines a name, most of the times that would be sufficient. But maybe keeping the original indices would be nice indeed!